tests/pos-macros/i12948b.scala

```diff
@@ -0,0 +1,8 @@
+import scala.quoted.*
+
+package foo {
+  trait Bar:
+    def bar(using Quotes) = '{ Baz }
```

Comment:
This is failing because we generate the inline accessor for `Baz` in this class.
```scala
trait Bar:
  def bar(using Quotes) = '{ this.foo$Bar$$inline$Baz }
  def foo$Bar$$inline$Baz = Baz
```
which then fails because it tries to access the `this` in the quote
```scala
5 |    def bar(using Quotes) = '{ Baz }
  |                               ^
  |                              access to Bar.this from wrong staging level:
  |                               - the definition is at level 0,
  |                               - but the access is at level 1.
```
We should have the inline accessor placed in the same scope as `Baz` to make its reference stage consistent.